src/jdk.httpserver/share/classes/sun/net/httpserver/ServerImpl.java

```diff
@@ -223,44 +225,69 @@ public HttpsConfigurator getHttpsConfigurator () {
     }
 
     public final boolean isFinishing() {
-        return finished;
+        return finishedLatch.getCount() == 0;
     }
 
+    /**
+     * This method stops the server by adding a stop request event and
+     * waiting for the server until the event is triggered or until the maximum delay is triggered.
+     * <p>
+     * This ensures that the server is stopped immediately after all exchanges are complete. HttpConnections will be forcefully closed if active exchanges do not
+     * complete within the imparted delay.
+     *
+     * @param delay maximum delay to wait for exchanges completion, in seconds
+     */
     public void stop (int delay) {
         if (delay < 0) {
```

Comment:
with the use of CountDownLatch the negative check becomes superfluous i.e.

public boolean await(long timeout, TimeUnit unit) throws InterruptedException
Causes the current thread to wait until the latch has counted down to zero, unless the thread is interrupted, or the specified waiting time elapses.

. . . 
If the specified waiting time elapses then the value false is returned. If the time is less than or equal to zero, the method will not wait at all.

Reply:
It's not something we should change though, as it's part of the API contract in `HttpServer::stop`:
https://docs.oracle.com/en/java/javase/24/docs/api/jdk.httpserver/com/sun/net/httpserver/HttpServer.html#stop(int)